neo3/wallet/wallet.py

```diff
@@ -0,0 +1,44 @@
+import json
+import os.path
+from .scrypt_parameters import ScryptParameters
+
+
+def encode_scrypt_parameters(scrypt_parameters):
+    if isinstance(scrypt_parameters, ScryptParameters):
+        return {'n': scrypt_parameters.n, 'r': scrypt_parameters.r, 'p': scrypt_parameters.p}
+    else:
+        type_name = scrypt_parameters.__class__.__name__
+        raise TypeError(f"Object of type '{type_name}' is not JSON serializable")
+
+
+class Wallet:
+
+    def __init__(self, path=None, name=None, json_wallet=None):
+        if json_wallet is not None:
+            self.load_json(json_wallet)
+        else:
+            self.path = path
+
+            if os.path.isfile(path):
+                with open(path) as json_file:
+                    data = json.load(json_file)
+                self.load_json(data)
+            else:
+                self.name = name
+                self.version = '3.0'
+                self.scrypt = ScryptParameters.default()
+                self.accounts = {}
```

Comment:
Should this be a list instead of a dictionary? `test_wallet.py` line 19 tests for `[]`.